narwhals/_pandas_like/dataframe.py

```diff
@@ -277,6 +278,30 @@ def join(
         if isinstance(right_on, str):
             right_on = [right_on]
 
+        if how == "cross":
+            if self._implementation in {"modin", "cudf"} or (
+                self._implementation == "pandas"
+                and (pd := get_pandas()) is not None
+                and parse_version(pd.__version__) < parse_version("1.2.0")
+            ):
+                return self._from_dataframe(
+                    self._dataframe.assign(__cross_join_key__=0).merge(
+                        other._dataframe.assign(__cross_join_key__=0),
+                        how="inner",
+                        left_on="__cross_join_key__",
+                        right_on="__cross_join_key__",
+                        suffixes=("", "_right"),
+                    ),
+                ).drop("__cross_join_key__")
```

Comment:
@MarcoGorelli this is the workaround. Is it nice? Not really!

If someone comes with a dataframe with a column called `__cross_join_key__` I will get very sad. Maybe we can do even one more step further with `__narwhals_cross_join_key__`?!

Reply:
How about generating a random token until we find a string which isn't already a column name?

I did something [here](https://github.com/psf/black/pull/2357/files) once, check `get_token` in `src/black/handle_ipynb_magics.py`